language/move-prover/tests/sources/restrictions.move

```diff
@@ -0,0 +1,60 @@
+// no-boogie-test
+module M {
+
+    struct S {
+        f: u64
+    }
+
+    spec module {
+        // Tuples as result type.
+        define f1(): (num, num) { (1, 2) }
+
+        // Functions as result type.
+        define f2(): | |num { | | 1 }
+
+        // Invoke
+        define f3(f: |num|num): num {
```

Comment:
f3 is also blowing my head. Is num an implicit type parameter to f3?

Reply:
Perhaps "num" is type of God's integers?

Reply:
Yes, its the arbitrary precision number type which is also in the spec language doc.